src/compiler/compiler.rs

```diff
@@ -76,6 +76,15 @@ pub enum CompilerKind {
     Rust,
 }
 
+impl CompilerKind {
+    pub fn lang_kind(&self) -> String {
```

Comment:
I would probably just make this return `&'static str` and have the caller deal with turning it into a `String` (you're already calling `clone` on it there anyway).